packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderApiKeyAuthTests.cs

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+using Microsoft.Generator.CSharp.Tests.Common;
+using NUnit.Framework;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Tests.Providers.ClientProviders
+{
+    public class ClientProviderApiKeyAuthTests
+    {
+        private const string SubClientsCategory = "WithSubClients";
+        private const string TestClientName = "TestClient";
+        private static readonly InputClient _animalClient = new("animal", "AnimalClient description", [], [], TestClientName);
+        private static readonly InputClient _dogClient = new("dog", "DogClient description", [], [], _animalClient.Name);
+        private static readonly InputClient _huskyClient = new("husky", "HuskyClient description", [], [], _dogClient.Name);
+
+        [SetUp]
+        public void SetUp()
+        {
+            var categories = TestContext.CurrentContext.Test?.Properties["Category"];
+            bool containsSubClients = categories?.Contains(SubClientsCategory) ?? false;
+
+            if (containsSubClients)
+            {
+                MockHelpers.LoadMockPlugin(
+                    apiKeyAuth: () => new InputApiKeyAuth("mock", null),
+                    clients: () => [_animalClient, _dogClient, _huskyClient]);
+            }
+            else
+            {
+                MockHelpers.LoadMockPlugin(apiKeyAuth: () => new InputApiKeyAuth("mock", null));
+            }
+        }
+
+        [Test]
+        public void TestBuildProperties()
+        {
+            var client = InputFactory.Client(TestClientName);
+            var clientProvider = new ClientProvider(client);
+
+            Assert.IsNotNull(clientProvider);
+
+            // validate the properties
+            var properties = clientProvider.Properties;
+            Assert.IsTrue(properties.Count > 0);
+            // there should be a pipeline property
+            Assert.AreEqual(1, properties.Count);
+
+            var pipelineProperty = properties.First();
+            Assert.AreEqual(typeof(ClientPipeline), pipelineProperty.Type.FrameworkType);
+            Assert.AreEqual("Pipeline", pipelineProperty.Name);
+            Assert.AreEqual(MethodSignatureModifiers.Public, pipelineProperty.Modifiers);
+        }
+
+        [TestCaseSource(nameof(BuildFieldsTestCases))]
+        public void TestBuildFields(List<InputParameter> inputParameters, bool containsAdditionalParams)
```

Comment:
Same question here - I understand that we want to validate that the auth fields are present, but I think it would be more clear to just validate the specific auth field in this test, and leave the general test in ClientProviderTests.

Reply:
well - I am struggling on this as well.
I added a few more classes in my new PR: ClientProviderApiKeyAuthTests, ClientProviderNoAuthTests, ClientProviderOAuth2AuthTests, ClientProviderBothAuthTests.
And it is not quite simple that we could predicate which one should we test, therefore in my new PR, I am validating ALL fields to be expected.
Would this be OK for you?
Maybe I should migrate those changes here as well

Reply:
More singularly focused unit tests are easier to understand and debug. I would prefer that the tests being added to the new test class are scoped to auth feature that we are adding.

Reply:
OK - so I will update my the other PR so that we will have one `ClientProviderAuthTests`.

And how about this:
ClientProviderTests will have a `TestBuildFields` to check if those optional/required client fields are present.
ClientProviderAuthTests will have a `TestBuildAuthFields` to check if those auth related fields are present.